Firestore/Source/Local/FSTMemoryPersistence.mm

```diff
@@ -236,9 +237,14 @@ - (int)removeTargetsThroughSequenceNumber:(ListenSequenceNumber)sequenceNumber
 }
 
 - (int)removeOrphanedDocumentsThroughSequenceNumber:(ListenSequenceNumber)upperBound {
-  return [(FSTMemoryRemoteDocumentCache *)_persistence.remoteDocumentCache
-      removeOrphanedDocuments:self
-        throughSequenceNumber:upperBound];
+  std::vector<DocumentKey> removed =
+      [(FSTMemoryRemoteDocumentCache *)_persistence.remoteDocumentCache
+          removeOrphanedDocuments:self
+            throughSequenceNumber:upperBound];
+  for (auto it = removed.begin(); it != removed.end(); it++) {
```

Comment:
`auto` only infers the type of value it's holding but does not infer qualifiers. Unfortunately, it doesn't mean "do the right thing for the context in which I'm using this type."

Unless you specifically need a copy of the thing thing you're iterating over, you should still make this a const reference:

```c++
for (const auto& it = ...) {
```

Here and above (sorry I missed this on the first pass).

Reply:
switched to `for (const auto& entry : ...)` here and elsewhere in the file.